tensorflow/core/kernels/mkl_conv_ops.h

```diff
@@ -351,15 +368,23 @@ class MklDnnConvUtil {
 
     // Output batch is same as input batch.
     int out_batch = GetTensorDim(input_shape, data_format_, 'N');
-
-    // Output depth is same as last dimension for filter.
-    int out_depth = filter_shape.dim_size(isConv2D ? 3 : 4);
+    int out_depth;
+    
+    // TODO add support for 3-D Depthwise
+
+    // Output depth is same as last dimension for filters for regular 
+    // convolutions. For depthwise it is in_depth*channel_muliplier 
+    if (is_Depthwise) {
+      out_depth = (filter_shape.dim_size(2)*filter_shape.dim_size(3));
```

Comment:
For future PRs, what do you think about adding an enum (for TF format indices) or getter functions? For example, `filter_shape.dim_size(TFFilterDims::in_depth)` or `GetInDepthFromTFFormat(filter_shape)`. It will be more readable and flexible for future changes.